presto-hive/src/main/java/com/facebook/presto/hive/CachingFileOpener.java

```diff
@@ -0,0 +1,41 @@
+package com.facebook.presto.hive;
+
+import com.facebook.presto.cache.CacheManager;
+import com.facebook.presto.cache.CachingFileSystem;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public class CachingFileOpener
+        implements FileOpener
+{
+    private CacheManager cacheManager;
+    private FileOpener fileOpener;
+
+    @Inject
+    public CachingFileOpener(
+            CacheManager cacheManager,
+            @ForWarmStorageAwareFileOpener FileOpener fileOpener)
```

Comment:
nit: After renaming this should be changed as well, but still leave a comment just in case.